tests/tests/buffer.rs

```diff
@@ -326,3 +326,62 @@ static MINIMUM_BUFFER_BINDING_SIZE_DISPATCH: GpuTestConfiguration = GpuTestConfi
             let _ = encoder.finish();
         });
     });
+
+#[gpu_test]
+static CLEAR_OFFSET_OUTSIDE_RESOURCE_BOUNDS: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(TestParameters::default().test_features_limits())
+    .run_sync(|ctx| {
+        let size = 16;
+
+        let buffer = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+            label: None,
+            size,
+            usage: wgpu::BufferUsages::COPY_DST,
+            mapped_at_creation: false,
+        });
+
+        let out_of_bounds = size.checked_add(wgpu::COPY_BUFFER_ALIGNMENT).unwrap();
+
+        ctx.device.push_error_scope(wgpu::ErrorFilter::Validation);
+        ctx.device
+            .create_command_encoder(&Default::default())
+            .clear_buffer(&buffer, out_of_bounds, None);
+        let err_msg = pollster::block_on(ctx.device.pop_error_scope())
+            .unwrap()
+            .to_string();
+        assert!(err_msg.contains(
+            "Clear of 20..20 would end up overrunning the bounds of the buffer of size 16"
+        ));
```

Comment:
Could we work this into the `fail` function in some way?

Reply:
Ah, I actually intended to bring this up (though as follow-up work) and had forgotten about it. 😅 It seems like narrowing the specificity of a validation error would give a lot more confidence in a test being what we want. Do you consider this blocking for this PR? 🤔

While I'm thinking about it: I think adding a third argument is a good solution, unless we want to smith a one-off `Option`-alike with more descriptive variant names. I think confronting test authors with "please consider adding a message" is a good thing; I'd prefer to avoid a builder for this one.

Reply:
Filed a separate PR: https://github.com/gfx-rs/wgpu/pull/5282